src/commands/force/mdapi/beta/convert.ts

```diff
@@ -0,0 +1,215 @@
+
+  public async run(): Promise<ConvertCommandResult> {
+    await this.convert();
+    this.resolveSuccess();
+    return this.formatResult();
+  }
+
+  protected async convert(): Promise<void> {
+    this.rootDir = this.resolveRootDir(this.getFlag<string>('rootdir'));
+    this.outputDir = this.resolveOutputDir(this.getFlag<string>('outputdir'));
+    const metadatapath = this.resolveMetadataPaths(this.getFlag<string[]>('metadatapath'));
+    const manifest = this.resolveManifest(this.getFlag<string>('manifest'));
+    const metadata = this.getFlag<string[]>('metadata');
+
+    let paths: string[];
+    if (metadatapath) {
+      paths = metadatapath;
+    } else if (!manifest && !metadata) {
+      paths = [this.rootDir];
+    }
+
+    this.componentSet = await ComponentSetBuilder.build({
+      sourcepath: paths,
+      manifest: manifest && {
+        manifestPath: manifest,
+        directoryPaths: [this.rootDir],
+      },
+      metadata: metadata && {
+        metadataEntries: metadata,
+        directoryPaths: [this.rootDir],
+      },
+    });
+
+    const numOfComponents = this.componentSet.getSourceComponents().toArray().length;
+    if (numOfComponents > 0) {
+      this.ux.startSpinner(`Converting ${numOfComponents} metadata components`);
+
+      const converter = new MetadataConverter();
+      this.convertResult = await converter.convert(this.componentSet, 'source', {
+        type: 'directory',
+        outputDirectory: this.outputDir,
+        genUniqueDir: false,
+      });
+      this.ux.stopSpinner();
+    }
+  }
+
+  // No-op.  Any failure would throw an error.  If no error, it's successful.
+  // The framework provides this behavior.
+  /* eslint-disable-next-line @typescript-eslint/no-empty-function */
+  protected resolveSuccess(): void {}
+
+  protected formatResult(): ConvertCommandResult {
+    const formatter = new ConvertResultFormatter(this.logger, this.ux, this.convertResult);
+
+    if (!this.isJsonOutput()) {
+      formatter.display();
+    }
+    return formatter.getJson();
+  }
+
+  private ensureFlagPath(options: EnsureFlagOptions): void {
+    const { flagName, path, type, throwOnENOENT } = options;
+    try {
+      const stats = fs.statSync(path);
+      if (type !== 'any') {
+        const isDir = stats.isDirectory();
+        if ((type === 'dir' && !isDir) || (type === 'file' && isDir)) {
+          throw SfdxError.create('@salesforce/plugin-source', 'md.convert', 'InvalidFlagPath', [flagName, path]);
+        }
+      }
+    } catch (error: unknown) {
+      const err = error as FsError;
+      if (err.code !== 'ENOENT') {
+        throw err;
+      } else {
+        if (throwOnENOENT) {
+          throw SfdxError.create('@salesforce/plugin-source', 'md.convert', 'InvalidFlagPath', [flagName, path]);
+        }
+        fs.mkdirSync(path, { recursive: true });
+      }
+    }
+  }
+
+  private resolveRootDir(rootDir: string): string {
+    const resolvedRootDir = resolve(rootDir.trim());
```

Comment:
why not do the trimming either in `ensureFlagPath` or in the flags themselves via `flag.parse` instead of in each of these?

Reply:
Yeah that's a good point.  I think either place could work.  I'll work on that.